packages/playwright/src/reporters/base.ts

```diff
@@ -489,11 +490,35 @@ export function stripAnsiEscapes(str: string): string {
   return str.replace(ansiRegex, '');
 }
 
+function characterWidth(c: string) {
+  return getEastAsianWidth.eastAsianWidth(c.codePointAt(0)!);
+}
+
+function stringWidth(v: string) {
+  let width = 0;
+  for (const { segment } of new Intl.Segmenter(undefined, { granularity: 'grapheme' }).segment(v))
```

Comment:
Sounds like this is no better than `for (const segment of v)` in your use case.

Reply:
I think it is:

```js
> [...new Intl.Segmenter(undefined, { granularity: 'grapheme' }).segment("👟")]
[ { segment: '👟', index: 0, input: '👟' } ]
> "👟".split("")
[ '\ud83d', '\udc5f' ]
```

Reply:
It makes a difference for grapheme clusters, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Symbol.iterator#description